aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -98,6 +101,46 @@ Tensor _dim_arange(const Tensor& like, int64_t dim) {
   return at::arange(like.size(dim), like.options().dtype(at::kLong));
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~ complex / complex_polar ~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tensor& complex_out(Tensor& result, const Tensor& real, const Tensor& imag) {
+  // Sort of hacky, but necessary so that input dtypes don't get promoted to complex.
+  auto iter = TensorIterator::comparison_op(result, real, imag,
+    /*check_mem_overlap=*/true);
+  complex_stub(iter.device_type(), iter);
+  return result;
+}
+
+Tensor complex(const Tensor& real, const Tensor& imag) {
+  c10::TensorOptions options = real.options();
+  switch (promote_types(real.scalar_type(), imag.scalar_type())) {
```

Comment:
I think @anjali411 wrote a function that maps float types to their corresponding complex types.

Should this throw a runtime error if given anything other than float32 or float64 tensors? You can just require real and imag have the same dtype, too.